### PR TITLE
[sc-7298] add analytics

### PR DESCRIPTION
**Pull request by steb-veksler**

### CODE REVIEW

File: pkg/segment/user.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package segment
+
+import "github.com/segmentio/analytics-go/v3"
+
+const ORG_TARIAT_NAME = "orgName"
```
Comment (nitzangr) [inline pkg/segment/user.go:5]: TRAIT

File: pkg/segment/user.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+var userId string
+
+func NewUser(email string, org string) error {
```
Comment (nitzangr) [inline pkg/segment/user.go:9]: If no ones checks for the returning error, don't return it

File: pkg/segment/user.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	orgGroup := analytics.Group{
+		GroupId: org,
+		UserId:  user.UserId,
+		Traits:  analytics.NewTraits().SetName(org),
+	}
+
+	if err = client.Enqueue(user); err != nil {
+		return err
```
Comment (nitzangr) [inline pkg/segment/user.go:24]: Should we `SetUser` if `Enqueue` fails?

File: main.go [hunk 4]
```diff
@@ -25,19 +27,24 @@ func main() {
...
-	sentryClientOptions := sentry_utils.GetSentryClientOptions(environment, release)
+	sentryClientOptions := sentry_utils.GetSentryClientOptions(APP_NAME, environment, cmd.BinaryVersion)
 	if err = sentry.Init(sentryClientOptions); err != nil {
 		ui.GlobalWriter.PrintErrorMessageln(err.Error())
 		panic(err)
 	}
 	defer sentry.Flush(sentry_utils.FLUSH_TIMEOUT)
 
+	segmentConfig := segment.GetConfig(APP_NAME, cmd.BinaryVersion)
+	if err = segment.Init(segmentConfig); err != nil {
```
Comment (nitzangr) [inline main.go:42]: Segment shouldn't fail the process

File: pkg/segment/client.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+var (
+	client   analytics.Client
+	WriteKey string = "FPPzr8mdiYq9Ry2YOEVFN751DvSdwwUZ"
+)
+
+func GetConfig(appName, version string) analytics.Config {
+	return analytics.Config{
+		BatchSize: 1,
+		Interval:  5 * time.Second,
```
Comment (nitzangr) [inline pkg/segment/client.go:17]: Is `Interval` relevant on `BatchSize == 1`?

**Review by nitzangr — CHANGES_REQUESTED** [review]

File: cmd/deploy.go [hunk 6]
```diff
@@ -215,7 +226,8 @@ func validateCluster(ctx context.Context, kubeClient *k8s.Client, namespace stri
 	}
 
 	if !clusterReport.IsCompatible {
-		return errors.New("can't continue with installation, cluster is not compatible for installation. Check solutions suggested by the CLI")
+		err = errors.New("can't continue with installation, cluster is not compatible for installation. Check solutions suggested by the CLI")
```
[the remark this answers is not in the document]
Comment (steb-veksler) [reply cmd/deploy.go:229]: needed for defer to catch the error

File: main.go (hunk 4 above)
Comment (steb-veksler) [reply main.go:42]: The function will return an error if the configuration contained impossible, so needs to panic here

**Review by nitzangr — APPROVED** [review]

**Pull request merged**